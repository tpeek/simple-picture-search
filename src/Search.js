import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  padding: {
    margin: 8,
  },
  base: {
    margin: '16px auto',
    minWidth: 50,
    maxWidth: 350,
  },
  button: {
    textTransform: 'none',
    margin: 8,
    backgroundColor: '#3e00d5',
    height: 56,
    fontSize: 20,
  },
});

const Search = ({ onSearch }) => {
  const classes = useStyles();
  const [keyword, setKeyword] = useState('');
  const [category, setCategory] = useState('');

  const handleKeywordChange = (e) => {
    setKeyword(e.target.value);
  };

  const handleCategoryChange = (e) => {
    setCategory(e.target.value);
  };

  return (
    <div className={classes.base}>
      <TextField
        label="Keyword..."
        variant="outlined"
        fullWidth
        className={classes.padding}
        onChange={handleKeywordChange}
        value={keyword}
      />
      <TextField
        select
        variant="outlined"
        label="Category..."
        fullWidth
        className={classes.padding}
        onChange={handleCategoryChange}
        value={category}
      />
      <Button
        color="primary"
        variant="contained"
        size="large"
        fullWidth
        className={classes.button}
        classes={{ root: classes.buttonCasing }}
        onClick={() => onSearch(keyword, category)}
      >
        Search
      </Button>
    </div>
  );
};

export default Search;
