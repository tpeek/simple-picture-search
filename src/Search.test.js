import React from 'react';
import Search from './Search';

const defaultProps = {
  onSearch: jest.fn(),
};

const setup = (props) => shallow(<Search {...defaultProps} {...props} />);

describe('<Search />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
