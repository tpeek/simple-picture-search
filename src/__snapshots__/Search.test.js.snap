// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders snapshot correctly 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    className="makeStyles-spacing-2"
    fullWidth={true}
    label="Keyword..."
    onChange={[Function]}
    value=""
    variant="outlined"
  />
  <WithStyles(ForwardRef(TextField))
    className="makeStyles-spacing-2"
    fullWidth={true}
    label="Category..."
    onChange={[Function]}
    select={true}
    value=""
    variant="outlined"
  >
    <WithStyles(ForwardRef(MenuItem))
      key="Animals"
      value="animals"
    >
      Animals
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Backgrounds"
      value="backgrounds"
    >
      Backgrounds
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Buildings"
      value="buildings"
    >
      Buildings
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Business"
      value="business"
    >
      Business
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Computer"
      value="computer"
    >
      Computer
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Education"
      value="education"
    >
      Education
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Fashion"
      value="fashion"
    >
      Fashion
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Feelings"
      value="feelings"
    >
      Feelings
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Food"
      value="food"
    >
      Food
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Health"
      value="health"
    >
      Health
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Industry"
      value="industry"
    >
      Industry
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Music"
      value="music"
    >
      Music
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Nature"
      value="nature"
    >
      Nature
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="People"
      value="people"
    >
      People
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Places"
      value="places"
    >
      Places
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Religion"
      value="religion"
    >
      Religion
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Science"
      value="science"
    >
      Science
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Sports"
      value="sports"
    >
      Sports
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Transportation"
      value="transportation"
    >
      Transportation
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Travel"
      value="travel"
    >
      Travel
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(TextField))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-3"
    color="primary"
    fullWidth={true}
    size="large"
    type="submit"
    variant="contained"
  >
    Search
  </WithStyles(ForwardRef(Button))>
</form>
`;
